### DISCUSSION TITLE
JSON::Object stringify unicode to UTF8 format but parser can not handle it

### DISCUSSION THREAD

**Discussion by Lloyd-Pottiger** [pocoproject/poco#4707] [Q&A]
**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior. If feasible, please provide a [SSCCE](http://www.sscce.org/) - usually, it will make the process much easier and faster.

```cpp
TEST(JSONParser, TestUnicode)
try
{
    String json_str = R"json({"default":"\u0050\u0050"})json";
    Poco::JSON::Parser parser;
    Poco::Dynamic::Var result = parser.parse(json_str);
    const auto & obj = result.extract<Poco::JSON::Object::Ptr>();
    auto default_val = obj->get("default");
    Poco::JSON::Object::Ptr json = new Poco::JSON::Object();
    json->set("default", default_val);
    std::stringstream ss;
    json->stringify(ss);
    std::cout << ss.str() << std::endl;
    ASSERT_TRUE(ss.str() == json_str);
}
CATCH
```

**Expected behavior**

**Logs**
If applicable, add logs to help explain your problem.

**Screenshots**

**Please add relevant environment information:**
 - OS Type and Version
 - POCO Version
 - Third-party product (eg. database or library) type and version

**Additional context**

**Comment by micheleselea** [comment]
This string is not UTF8 string, that is an escaped string
String json_str = R"json({"default":"\u0050\u0050"})json";

you have to unescape before to obtain the UTF8 string, something like
json_str_unescape = Poco::UTF8::unescape(json_str)

**Comment by Lloyd-Pottiger** [comment]
@micheleselea It seems not to work for the following case:

```cpp
TEST(JSONParser, TestUnicode)
try
{
    String json_str = R"json({"default":"\u0007\u0007"})json";
    auto json_str_unescape = Poco::UTF8::unescape(json_str);
    Poco::JSON::Parser parser;
    Poco::Dynamic::Var result = parser.parse(json_str);
    const auto & obj = result.extract<Poco::JSON::Object::Ptr>();
    auto default_val = obj->get("default");
    Poco::JSON::Object::Ptr json = new Poco::JSON::Object();
    json->set("default", default_val);
    std::stringstream ss;
    json->stringify(ss);
    std::cout << ss.str() << std::endl;
    ASSERT_TRUE(ss.str() == json_str_unescape);
}
CATCH
```

The json generated by https://go.dev/play/p/6o6kX0vOIsW

**Comment by aleks-f** [comment]
@Lloyd-Pottiger of course it works. I'm not quite sure what the complaint is here. I'l move it to discussions

<img width="709" alt="image" src="https://github.com/user-attachments/assets/d06482a3-dee4-4a32-971b-2a967a70dae9">

**Comment by aleks-f** [comment]
and this one works, too:

<img width="706" alt="image" src="https://github.com/user-attachments/assets/a222a01e-5c5d-4d95-accf-76814a1a1d53">

**Comment by Lloyd-Pottiger** [comment]
@aleks-f I found my poco version is too old, after upgrading to the latest, it works. Thanks very much~
